pyro/distributions/random_primitive.py

```diff
@@ -34,7 +36,8 @@ def shape(self, x=None, *args, **kwargs):
         return self.dist_class(*args, **kwargs).shape(x)
 
     def sample(self, *args, **kwargs):
-        return self.dist_class(*args, **kwargs).sample()
+        sample_shape = kwargs.pop('sample_shape', torch.Size())
```

Comment:
I think you'll need to `kwargs.pop('sample_shape', None)` in all other methods too, otherwise this will be passed to the distribution constructor.

Reply:
That's right, this is being absorbed into args/kwargs in the constructor, but will change to remove `batch_shape` in the methods here.

Reply:
Makes sense.